pkg/util/storage/manager.go

```diff
@@ -44,6 +46,22 @@ func (m *manager) BlobService(ctx context.Context, resourceGroup, account string
 		SharedAccessExpiryTime: &date.Time{Time: t.Add(24 * time.Hour)},
 	})
 	if err != nil {
+		if detailedError, ok := err.(autorest.DetailedError); ok {
+			if detailedError.StatusCode == http.StatusTooManyRequests {
+				err = &api.CloudError{
+					StatusCode: http.StatusTooManyRequests,
+					CloudErrorBody: &api.CloudErrorBody{
+						Code:    api.CloudErrorCodeThrottlingLimitExceeded,
+						Message: "ThrottlingLimitExceeded",
+						Details: []api.CloudErrorBody{
+							{
+								Message: string("Requests are being throttled due to Azure Storage limits being exceeded. Please visit https://learn.microsoft.com/en-us/azure/openshift/troubleshoot#exceeding-azure-storage-limits for more details."),
+							},
+						},
+					},
+				}
+			}
+		}
```

Comment:
I think we could extract that to an auxiliar function and leverage the concept of early return to simplify the levels of nesting of the block of code. It is pretty common in Go to exit as soon as possible and I think we can take advantage of it 🙂 
Line 65 could become 
```go
return nil, getCorrectErrWhenTooManyRequests(err)
```
and getCorrectErrWhenTooManyRequests would be something like
```go
func getCorrectErrWhenTooManyRequests(err error) error {
	detailedError, ok := err.(autorest.DetailedError)
	if !ok {
		return err
	}

	if detailedError.StatusCode != http.StatusTooManyRequests {
		return err
	}

	msg := "Requests are being throttled due to Azure Storage limits being exceeded. Please visit https://learn.microsoft.com/en-us/azure/openshift/troubleshoot#exceeding-azure-storage-limits for more details."
	cloudError := &api.CloudError{
		StatusCode: http.StatusTooManyRequests,
		CloudErrorBody: &api.CloudErrorBody{
			Code:    api.CloudErrorCodeThrottlingLimitExceeded,
			Message: "ThrottlingLimitExceeded",
			Details: []api.CloudErrorBody{
				{
					Message: string(msg),
				},
			},
		},
	}

	return cloudError
}
```

Reply:
updated the code with the mentioned changes..